doc/cephadm/services/mgmt-gateway.rst

```diff
@@ -0,0 +1,136 @@
+* ``High Availability for monitoring``: nginx HA mechanisms are used to provide high availability for monitoring.
+
+Security enhancements
+=====================
+
+Once the `mgmt-gateway` service is deployed user cannot access monitoring services without authentication through the
+dashboard.
+
+
+High availability enhancements
+==============================
+nginx HA mechanisms are used to provide high availability for all the Ceph management applications including Dashboard
+and monitoring stack. In case of the Dashboard user shouldn't care anymore where the active manager is running.
+`mgmt-gateway` handles manager failover transparently and redirects the user to the active manager. In case of the
+monitoring `mgmt-gateway` takes care of handling the high availability in case several instances of Prometheus, Alertmanager
+or Grafana are available. The reverse-proxy will automatically detect healthy instances and use them to process user requests.
+
+
+Accessing services with mgmt-gateway
+====================================
+
+Once the `mgmt-gateway` service is deployed direct access to the monitoring services will not be allowed anymore.
+Applications such as: Prometheus, Grafana and Alertmanager are now accessible through dedicated links
+from `Administration > Services`.
+
+
+Service Specification
+=====================
+
+An mgmt-gateway service can be applied using a specification. An example in YAML follows:
+
+.. code-block:: yaml
+
+    service_type: mgmt-gateway
+    service_id: gateway
+    placement:
+      hosts:
+        - ceph0
+    spec:
+     port: 5000
+     ssl_protocols:
+       - TLSv1.2
+       - TLSv1.3
+       - ...
+     ssl_ciphers:
+       - AES128-SHA
+       - AES256-SHA
+       - ...
+     ssl_certificate: |
+       -----BEGIN CERTIFICATE-----
+       MIIDtTCCAp2gAwIBAgIYMC4xNzc1NDQxNjEzMzc2MjMyXzxvQ7EcMA0GCSqGSIb3
+       DQEBCwUAMG0xCzAJBgNVBAYTAlVTMQ0wCwYDVQQIDARVdGFoMRcwFQYDVQQHDA5T
+       [...]
+       -----END CERTIFICATE-----
+    ssl_certificate_key: |
+       -----BEGIN PRIVATE KEY-----
+       MIIEvQIBADANBgkqhkiG9w0BAQEFAASCBKcwggSjAgEAAoIBAQC5jdYbjtNTAKW4
+       /CwQr/7wOiLGzVxChn3mmCIF3DwbL/qvTFTX2d8bDf6LjGwLYloXHscRfxszX/4h
+       [...]
+       -----END PRIVATE KEY-----
+
+Fields specific to the ``spec`` section of the mgmt-gateway service are described below.
+
+.. py:currentmodule:: ceph.deployment.service_spec
+
+.. autoclass:: MgmtGatewaySpec
+   :members:
+
+The specification can then be applied by running the following command:
+
+.. prompt:: bash #
+
+   ceph orch apply -i mgmt-gateway.yaml
+
+
+Limitations
+===========
+
+A non-exhaustive list of important limitations for the mgmt-gateway service follows:
+
+* High-availability configurations and clustering for the mgmt-gateway service are currently not supported.
+* Services must bind to the appropriate ports based on the applications being proxied. Ensure that there
+  are no port conflicts that might disrupt service availability.
```

Comment:
Does this mean that nginx and the origin services can't run on the same nodes?  Or do you expose a non-default port for each?

Reply:
> Does this mean that nginx and the origin services can't run on the same nodes? Or do you expose a non-default port for each?

By default the service will use default ports (443 for https / 80 when https is disabled). So this sentence just to warn about the port conflict risk that exists if the user doesn't specify different port. Anway service port is configurable in the spec so the user can change it to listen on any other valid port.